devU-api/src/containerAutoGrader/containerAutoGrader.router.ts

```diff
@@ -0,0 +1,52 @@
+import express from 'express';
+import multer from 'multer';
+
+import validator from './containerAutoGrader.validator';
+import { asInt } from '../middleware/validator/generic.validator';
+
+import ContainerAutoGraderController from './containerAutoGrader.controller';
+
+const Router = express.Router();
+const upload = multer();
+
+/**
+ * @swagger
+ * /container-auto-graders:
+ *   get:
+ *     summary: Retrieve a list of all container auto graders
+ */
+Router.get('/', ContainerAutoGraderController.get);
+
+/**
+ * @swagger
+ * /container-auto-graders/{id}:
+ *   get:
+ *     summary: Retrieve a single container auto grader
+ */
+Router.get('/:id', asInt(), ContainerAutoGraderController.detail);
+
+/**
+ * @swagger
+ * /container-auto-graders:
+ *   post:
+ *     summary: Create a new container auto grader
+ */
+Router.post('/', upload.fields([{name: 'graderFilename'},{name: 'makefileFilename'}]), validator, ContainerAutoGraderController.post);
+
+/**
+ * @swagger
+ * /container-auto-graders/{id}:
+ *   put:
+ *     summary: Update a container auto grader's grader file and/or makefile
```

Comment:
can also update timeout, autograding image, and any other fields we may add in the future. Just leave this at update a container auto grader